torch/_inductor/decomposition.py

```diff
@@ -962,56 +963,86 @@ def index_reduce(
     )
 
 
-@register_decomposition(aten.max_pool2d_with_indices)
-def max_pool2d_with_indices(
+def _max_pool_with_indices(
     x: torch.Tensor,
     kernel_size: list[int],
-    stride: Optional[Union[int, list[int]]] = None,
-    padding: Union[int, list[int]] = 0,
-    dilation: Union[int, list[int]] = 1,
-    ceil_mode: bool = False,
+    stride: Optional[Union[int, list[int]]],
```

Comment:
when would stride be Optional ?

Reply:
When stride is None, it becomes kernel_size. This is the signature of the aten operation.
```
In [4]: aten.max_pool2d_with_indices.default._schema
Out[4]: aten::max_pool2d_with_indices(Tensor self, int[2] kernel_size, int[2] stride=[], int[2] padding=0, int[2] dilation=1, bool ceil_mode=False) -> (Tensor, Tensor)
```